src/client/InteractionClient.js

```diff
@@ -0,0 +1,221 @@
+'use strict';
+
+const BaseClient = require('./BaseClient');
+const ApplicationCommand = require('../structures/ApplicationCommand');
+const CommandInteraction = require('../structures/CommandInteraction');
+const { Events, ApplicationCommandOptionType, InteractionType, InteractionResponseType } = require('../util/Constants');
+const MessageFlags = require('../util/MessageFlags');
+let sodium;
+
+function transformCommand(command) {
+  return {
+    ...command,
+    options: command.options.map(function m(o) {
+      return {
+        ...o,
+        type: ApplicationCommandOptionType[o.type],
+        options: o.options?.map(m),
+      };
+    }),
+  };
+}
+
+/**
+ * Interaction client is used for interactions.
+ *
+ * @example
+ * const client = new InteractionClient({
+ *   token: ABC,
+ *   publicKey: XYZ,
+ * });
+ *
+ * client.on('interactionCreate', (interaction) => {
+ *   // automatically handles long responses
+ *   if (will take a long time) {
+ *     interaction.defer();
+ *     doSomethingLong.then((d) => {
+ *       interaction.webhook.send({
+ *         content: `fancy data: ${d}`,
+ *       });
+ *     });
+ *   } else {
+ *     interaction.reply('hi!');
+ *   }
+ * });
+ * ```
+ */
+class InteractionClient extends BaseClient {
+  /**
+   * @param {Options} options Options for the client.
```

Comment:
this `Options` type doesn't exist, would probably be good to give it a better name too